src/pages/loading-waiting-states/progress-bar/index.tsx

```diff
@@ -0,0 +1,313 @@
+];
+const LinearProgressWithLabel = (props: LinearProgressProps & { value: number }): JSX.Element => (
+    <LinearProgress variant="determinate" {...props} />
+);
+const uploadFileList: FolderItem[] = [];
+
+const useStyles = makeStyles((theme: Theme) => ({
+    container: {
+        padding: `${theme.spacing(2)}px ${theme.spacing(2)}px`,
+        maxWidth: 600,
+        margin: '0 auto',
+    },
+    cardContent: {
+        padding: 0,
+        '&:last-child': {
+            paddingBottom: 0,
+        },
+    },
+    uploadButtonContainer: {
+        textAlign: 'right',
+        paddingBottom: `${theme.spacing(2)}px`,
+    },
+    closeButtonContainer: {
+        color: Colors.black[50],
+        borderColor: Colors.black[50],
+        [theme.breakpoints.up('sm')]: {
+            width: '80px',
+        },
+    },
+    formControl: {
+        width: '100%',
+    },
+    radioLabel: {
+        display: 'flex',
+    },
+    icon: {
+        fill: Colors.black[200],
+        marginLeft: `${theme.spacing(0.5)}px`,
+    },
+    iconContainer: {
+        marginRight: `${theme.spacing(2)}px`,
+        maxWidth: '40px',
+        minWidth: '40px',
+        width: '40px',
+        marginTop: `${theme.spacing(1)}px`,
+    },
+    formLabel: {
+        margin: 0,
+        width: '100%',
+        padding: theme.spacing(1),
+        borderBottom: `1px solid ${theme.palette.divider}`,
+        '&:last-child': {
+            borderBottom: 'none',
+        },
+    },
+    toolbarGutters: {
+        padding: `0 ${theme.spacing(2)}px`,
+    },
+    placementOfList: {
+        position: 'absolute',
+        bottom: theme.spacing(3),
+        right: theme.spacing(3),
+        [theme.breakpoints.down('xs')]: {
+            bottom: 0,
+            right: 0,
+            width: '100%',
+        },
+    },
+    infoList: {
+        marginBottom: theme.spacing(2),
+        '&:last-child': {
+            marginBottom: 0,
+        },
+    },
+    subTitle: {
+        color: Colors.black[200],
+    },
+    snackbar: {
+        position: 'inherit',
+        transform: 'none',
+    },
+    SnackbarContent: {
+        borderRadius: 0,
+        padding: `0 ${theme.spacing(2)}px 0 ${theme.spacing(1)}px`,
+    },
+    messageContainer: {
+        display: 'flex',
+        alignItems: 'center',
+    },
+    messageTextContainer: {
+        marginRight: `${theme.spacing(3)}px`,
+    },
+}));
+const createFileItem = (increment: number): FolderItem => ({
+    id: increment,
+    open: true,
+    duration: null,
+    buttonLabel: 'Close',
+    name: 'PX Blue is Awesome.pdf',
+    progress: 0,
+    status: `Uploading (0%)`,
+});
+
+export const ProgressBar = (): JSX.Element => {
+    const dispatch = useDispatch();
+    const classes = useStyles();
+    const [fileUploadList, setFileUploadList] = useState<FolderItem[]>(uploadFileList);
+
+    const [radioButtonvalue, setRadioButtonvalue] = useState('1');
+    const [count, setCount] = useState(0);
```

Comment:
Tried with array.length but if I am deleting more than 1 file and and files. It creates duplicate index.
![image](https://user-images.githubusercontent.com/10433274/125930647-a68df2e2-e58a-4d1d-b61b-aa506a022826.png)

Reply:
I see. That's because you are setting the id of the item based on the number of items in the list. If you just use a regular variable to track the unique index and always increment it, then it won't be tied to the length of the array at all and will always be unique:

like
```
let nextFileIndex = 0;
...
// when we click the button
uploadFile {
// use the next index and increment it
...createFile(nextFileIndex++);
}
```
and onDelete simply delete it...the id will never be reused if we just increment it each time we add a file